### PR TITLE
Implement deterministic GeoDataset

### LINKED ISSUE #1899: `random_bbox_assignment` non-deterministic behavior
### Description

The `random_bbox_assignment` dataset splitter (as well as all other spatial index-based splitters I suppose) is not deterministic between different program executions.

That is, splitting a dataset multiple times during the same runtime (e.g., by calling `GeoDataModule.setup()` multiple times in a row, always results in the same splits, which is expected. However, these splits are different when restarting the current kernel/program/script.

### Steps to reproduce

1. Go to the random `random_bbox_assignment` function definition and add the following line under `hits = list(dataset.index.intersection(dataset.index.bounds, objects=True))`: `x = [hit.object for hit in hits]`.
2. Insert a breakpoint in the following line (i.e., where `hits` is permuted).
3. Initialize the `L7IrishDataModule`.
4. Call `setup("fit")` on the data module and while debugging.
5. Catch the breakpoint.
6. Save the value of `x` somewhere.
7. Terminate the program.
8. Do Steps 3-6 again.
9. Notice that the value of `x` has changed.

### Version

0.5.1

### PR DESCRIPTION
**Pull request by DimitrisMantas**
Fixes #1899.

### CODE REVIEW

**Comment by DimitrisMantas** [comment]
@microsoft-github-policy-service agree

**Comment by DimitrisMantas** [comment]
OK, looking at this again about an hour later, I'd probably remove the implicit requirement for the file list to be ordered since the main thing we need to preserve is consistency.

So, maybe the order test can go, and so can the related fixture.

**Comment by DimitrisMantas** [comment]
On an unrelated note, `black` took way too long to run on my machine. Perhaps we should consider migrating to [ruff](https://github.com/astral-sh/ruff) at some point?

File: tests/datasets/test_geo.py [hunk 1]
```diff
@@ -76,6 +76,15 @@ class TestGeoDataset:
     def dataset(self) -> GeoDataset:
         return CustomGeoDataset()
 
+    @pytest.fixture(scope="class")
+    def files(self) -> list[str]:
```
Comment (adamjstewart) [inline tests/datasets/test_geo.py:80]: This definitely feels like overkill for what could just be a class variable.

Comment (DimitrisMantas) [reply tests/datasets/test_geo.py:80]: It's currently used in only one test, so maybe just move it there even?

**Review by adamjstewart — APPROVED** [review]
Looks perfect, thanks again for catching this bug and fixing it!

**Pull request merged**